src/telemetry-data/data-model.md

```diff
@@ -0,0 +1,203 @@
+* A measurement, a data-point, is defined by:
+  * A metric
+  * A timestamp
+  * A value
+* Meta-data might be attached to a metric:
+  * A group or a tag related to the source: a location, a device type, a serial number ...
+  * The unit of the measurements.
+* Meta-data might be attached to a measurement:
+  * The quality of the measurement (see e.g. [OPC Quality Code](https://honeywellprocess-community.force.com/opcsupport/s/article/What-are-the-OPC-Quality-Codes)).
+
+__Measurement collection over MQTT__ 
+
+Thin-edge use MQTT topics to move telemetry data around.
+* An MQTT topic
+  * can be specific to a metric (e.g. `collectd/raspberrypi/memory/used`)
+  * or multiplex several metrics (e.g. `tedge/measurements`).
+* The key attributes that define a metric (source and type) can be:
+  * implicit (any measurement published on `tedge/measurements` is coming from *the* device),
+  * given by the message payload (as the type is for `{ "temperature": 1085927424 }` ),
+  * given by the topic name (as for `collectd/raspberrypi/memory/used`).
+* The timestamp for a measurement can be:
+  * directly attached to the value (as for a collectd data point),
+  * share by a set of measurements (as in `{ "time": "2020-10-15T05:30:47+00:00", "temperature": 25, "pressure": 98 }`),
+  * or even implicit (and ambiguously defined as the current time of the consumer).
+
+To describe what is published on a topic, we need the following constituents:
+* The topic name and rules to extract metric attributes from it.
+* The message payload and rules to extract measurement attributes from it.
+* The values for any implied attributes.
+
+For instance, for collectd:
+
+```
+topic = 'collectd/raspberrypi/${metric.group}/${metric.name}'
+payload = CSV('${measurement.epoch}:${measurement.value}')
+implied = {
+   metric.source = device
+}
+```
+
+Or for thin-edge JSON 0.5:
+
+```
+topic = 'tedge/measurements/${metric.source}'
+payload = JSON('{ time: "${measurement.time}", "${metric.name}":${measurement.value} }')
+```
+
+## Message transformations
+
+A main goal is to be able to move data from one set of topics to another.
+For that one need to defined transformation that doesn't alter the semantics of the messages.
+The point is not necessarily to implement all these transformations but to leverage them.
+
+__Grouping measurements with a shared timestamp__
+
+The canonical thin-edge JSON message:
+
+```
+{
+    "time": T1,
+    "temperature": 25
+    "pressure": 98
+}
+```
+
+... can be decomposed into a list of messages that duplicate the timestamp:
+
+```
+[
+    {
+        "time": T1,
+    	"temperature": 25
+    },
+    {
+        "time": T1,
+    	"pressure": 98
+    }
+]
+```
+
+Vice-versa, the latter message can be transformed into the former:
+* provided the timestamps are the same,
+* or close enough (say within the same millisecond).
+
+__Grouping measurements with a shared metric group__
+
+The canonical thin-edge JSON multi-valued measurement:
```

Comment:
We really need to validate if any of our existing/potential customers find this extra level of hierarchy useful before we commit to supporting it. The feedback from last time was that not many of them find it useful.

Reply:
We have an indirect but important use-case with collectd where all the metrics are grouped.